### ISSUE TITLE
In error messages, only make the message red

### ISSUE THREAD

**Issue by DeborahK** [angular/angular#12044]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

Error messages appear with a massive amount of red text, which is very intimidating especially for new developers. They often take one look at it and then post questions for help instead of realizing that the message does contain the help they need.

**Expected behavior**

Consider only displaying the actual error message and its potential fixes in red. Display the file names, system messages and stack trace in some other color. That helps developers to focus in on the actual error and its recommended fix (if any).

**Minimal reproduction of the problem with instructions**

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **Angular version:** 2.0.X

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by vicb** [comment]
What are you trying to do ? What is the context ?

Please add a repro.

**Comment by DeborahK** [comment]
I'm trying to minimize the number of emails/questions I get from new Angular developers that are intimidated by the Angular error messages and therefore don't read them. Most of the Angular error messages very clearly state what the issue is and what needs to be done to solve them. But with *so* much red text, new developers especially are a bit overwhelmed and instead post questions on how to solve their problem.

**Comment by DeborahK** [comment]
Here's a screen shot of an example. Notice the amount of red text and how the actual problem is somewhat lost within all of the red.
![image](https://cloud.githubusercontent.com/assets/7987365/19043215/107f3ee0-8944-11e6-8dc8-e59f0f6a1ab5.png)

**Comment by pkozlowski-opensource** [comment]
Isn't just that we use [console.error](https://github.com/angular/angular/blob/a32078f85eaabe47d0abd4e8331d1d948bda2661/modules/%40angular/core/src/error_handler.ts#L55) and it is a browser does the rest of formatting (including coloring exception to red)?

I'm not sure how much we can do in terms of error msg formatting in the way that it works cross-browser (there are substantial differences in the way exception message is represented for example).

In short - it is a browser that makes things red. Doing custom formatting would be non-trivial.

**Comment by DeborahK** [comment]
Ah. So you don't have any way to add styles? Maybe we could accomplish something similar then with formatting? Adding the primary issue as the first line with a space before all of the detail?

**Comment by arciisine** [comment]
Looks like it might be possible to support in Firefox/Chrome without much effort. 
http://stackoverflow.com/questions/7505623/colors-in-javascript-console

**Comment by vicb** [comment]
@pkozlowski-opensource is right, it's only a `console.error()`

I don't think we want to implement browser specific behavior here (cf @arciisine) plus other people will probably start complaining it's not red enough...

**Comment by DeborahK** [comment]
"Not red enough"?
So you don't think that there are any issues with Angular developers (especially newer Angular developers) finding the actual error message within the page of red text?

**Comment by pkozlowski-opensource** [comment]
> So you don't think that there are any issues with Angular developers (especially newer Angular developers) finding the actual error message within the page of red text?

There is an issue of course as big blob of text representing an exception + its stack-trace might be hard to scan visually, especially for people new to web development. But:
* _any_ exception from _any_ JS code is logged this way by a browser so the issue is not specific to Angular and web developers should be already familiar with the way exception look
* trying to format exceptions in a non-standard way is very, very hard (and I mean really hard) in a cross-browser way
* trying to do custom formatting would mean shipping more code to a browser = slower startup time 

I don't think that @vicb is trying to say that things couldn't be better. It is just that "doing better" in this case is really hard and would mean more code shipped to a browser. So this is more of a tradeoff.

**Comment by arciisine** [comment]
Looking at something like https://github.com/adamschwartz/log, I think a simple polyfill could be added to console.log, to anticipate the angular stacktrace format.  I'm not sure if this belongs in core, but could easily be a tool only used during development, to keep production builds as small as possible.

**Comment by DeborahK** [comment]
Thank you @pkozlowski-opensource. That clarifies the reasoning.
Wish we could do more to help those that are getting into Web development because of Angular. And those many business devs that just need to get the job done quickly.
It seems that the Angular team spent time defining good and useful error messages. Its a shame we can't make them easier to find/see in the sea of red. :-(

**Comment by wardbell** [comment]
For whom is the zone.js output helpful?  I've never been able to make sense of it.

Why does the error for a missing HttpProvider begin "Uncaught (in promise): Error error"?

How can the user learn more about what "No provider for HTTP" means and what to do about it?

I don't think the answer to these questions is "we can't do anything about it".  I can think of better answers for at least (1) and (3). 

(1) Optionally suppress zone.js errors and stacktraces

(3) Error messages with links to web pages that can explain more.  Explaining more is a good docs task.

My REAL point is that we should be creative, not complacent. We have good evidence that the error message output is a significant impediment to understanding and adoption.. Can we proceed with (1) and (3)? If there are better ways, let's pursue them.

**Comment by pkozlowski-opensource** [comment]
@wardbell my comments about "possible" were only targeted at "making it less red". But of course there are many ways of making the entire messages different / better.

I would suggest opening small, focused issues for (1) and (3) as those seem to be 2 different things. Small focused issues are having _much_ better chance of getting solved as compared to one big "error message sucks" ticket that is about all and nothing actionable in the end.

I _do_ care a lot about error messages and [pushed number of commits to improve those. But let's be precise about what we want to improve.

**Comment by naomiblack** [comment]
Folks -- I'm going to lock comments for now -- let's discuss at the team meeting this afternoon.
